src/engine/threaded_engine.cc

```diff
@@ -403,7 +417,11 @@ inline void ThreadedEngine::OnComplete(ThreadedOpr* threaded_opr) {
   }
   // Mark complete for write variables.
   for (auto&& i : threaded_opr->mutable_vars) {
-    const bool debug_info = (engine_info_ && debug_wait_var_ == i);
+    if (threaded_opr->ex_ptr) {
+      i->ex_ptr = threaded_opr->ex_ptr;
+      if (!global_ex_ptr) global_ex_ptr = i->ex_ptr;
+    }
+    bool debug_info = (engine_info_ && debug_wait_var_ == i);
```

Comment:
why remove const?

Reply:
This was not intentional. Added it back.